### DISCUSSION TITLE
Support for HTML files in UTF8

### DISCUSSION THREAD

**Discussion by mplaut** [sumatrapdfreader/sumatrapdf#3557] [General]
I am having the following issue. The current version of Sumatra and also the new one, do not properly render HTML files that are UTF8 . The current version of mupdf does render these files properly. Is it possible to update to the current mupdf version?
I am attaching two test files, both with the same text. One is in Code page 1255 format which does not render properly in either Sumatra or in mupdf, and the other in UTF8 format which does render properly in mupdf but not in Sumatra. 
When I asked mupdf about supporting code pages I was told that it is too much work and anyway everything should be in UTF8. I can live with that and convert my old files, but Sumatra does not render the UTF8 files properly.
Can anything be done about that? Does it need a later version of mupdf?
[TEST.zip](https://github.com/sumatrapdfreader/sumatrapdf/files/11881263/TEST.zip)

TIA

**Comment by GitHubRulesOK** [comment]
the core issue is how the fonts may be handled since the result is expected to be American Times Roman `font-family:Times New` or `font-family:Times New Roman;` Which will work when the Page is shown as American text

![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/90bb931a-b668-4184-982c-020896a2c52a)

**Reply by mplaut** [reply]
Sumatra does display English text properly. But this is non-English text. In mupdf it renders properly as Hebrew text if it is encoded as UTF8. If it is encoded as code page 1255 it does not render properly in mupdf since, as someone told me, they did not implement code pages since there are too many of them, and anyway text should be in Unicode,. 
But then I did some more experimentation. It seems that even in Sumatra, if I define a Hebrew font, it will render correctly. Nonetheless, even the Times Roman font does have characters for the numbers defined for Hebrew. For some reason on Sumatra it does not render properly. In mupdf it does render properly. 
The thing is, I cannot be sure that everyone has a Hebrew defined font, so it would be better if it worked with Times Roman as well.
But this is definitely an improvement from my perspective.

**Reply by GitHubRulesOK** [reply]
Times Roman as a windows/dos default font is limited to very basic Ansi/Ascii codes thus the need to use wider UTF-8 ranges on Windows but its not well supported So MS used Segoe UI as its modern base whilst Google use more cross platform Generic Noto (I think that was used in Past by MuPDF lib)
SumatraPDF has used different default fonts at times but at some point some were found too "Heavyweight" for embedding. personally unsure which are retained and which discarded

**Reply by mplaut** [reply]
Thank you for your reply, but I am not sure if I follow completely. Using a font map program, I see that the Times Roman file in my Windows defines a lot of codes, including the Basic Hebrew that I am interested in. However when I use Times Roman as the font family, it does not render in Sumatra. When I use Tahoma or Frankruehl it does render in the proper font. In mupdf everything renders in correct Hebrew but it does not display the specific font called for. 
I tried Segoe UI and Noto and even Noto Hebrew and it did not render in Sumatra.
